itl/plugins-contrib.d/network-components.conf

```diff
@@ -833,3 +833,101 @@ object CheckCommand "printer_health" {
 	vars.printer_health_hostname = "$check_address$"
 	vars.printer_health_mode = "supplies-status"
 }
+
+object CheckCommand "fiberchannel-brocade" {
+	import "ipv4-or-ipv6"
+
+	command = [ PluginContribDir + "/check_snmp_brocade" ]
+
+	arguments = {
+		"--hostname" = {
+			value = "$fiberchannel_brocade_hostname$"
+			description = "Hostname or IP-address of the server (SNMP mode only)"
+			required = true
+		}
+		"--community" = {
+			value = "$fiberchannel_brocade_community$"
+			description = "SNMP community of the switch (SNMP v1/2 only)"
+		}
+		"--snmpversion" = {
+			value = "$fiberchannel_brocade_snmpversion$"
+			description = "The SNMP protocol to use (default: 2c, other possibilities: 1,3)"
+		}
+		"--seclevel" = {
+			value = "$fiberchannel_brocade_seclevel$"
+			description = "The SNMP security level to use (default: noAuthNoPriv, othet possibilities: authNoPriv,authPriv)"
+		}
+		"--port" = {
+			value = "$fiberchannel_brocade_port$"
+			description = "The SNMP port to use (default: 161)"
+		}
+		"--context" = {
+			value = "$fiberchannel_brocade_context$"
+			description = "SNMPv3 context name"
+		}
+		# It seems --fc-port parameter is not working, using its short form
+		"-P" = {
+			value = "$fiberchannel_brocade_fc_port$"
+			description = "Port number as shown in the output of `switchshow`. Can't combine with --systeminfo parameter"
+		}
+		"--systeminfo" = {
+			set_if = "$fiberchannel_brocade_systeminfo$"
+			description = "Get global data like boot date, overall status, reachability"
+		}
+		"--sensor" = {
+			set_if = "$fiberchannel_brocade_sensor$"
+			description = "Additional to --systeminfo. Status of powersupply, fans and temp sensors"
+		}
+		"--global" = {
+			set_if = "$fiberchannel_brocade_global$"
+			description = "Global data like boot date, overall status"
+		}
+		"--performancedata" = {
+			value = "$fiberchannel_brocade_perfdata$"
+			description = "Print performance data of the selected FC port specified via -P parameter"
+		}
+		"--secname" = {
+			value = "$fiberchannel_brocade_secname$"
+			description = "The SNMPv3 securityName for the USM security model"
+		}
+		"--authpassword" = {
+			value = "$fiberchannel_brocade_authpassword$"
+			description = "The authentication password for SNMPv3"
+		}
+		"--authproto" = {
+			value = "$fiberchannel_brocade_authprotocol$"
+			description = "The authentication protocol for SNMPv3 (md5|sha)"
```

Comment:
Again usage shows upper-case names.

Reply:
For the CLI parameter itself? Then I'd like it to be changed, _but not the macro_.

Reply:
Ah, great, Github shows the files in a different order in the conversation and files changed tabs. I'm referring to the algorithm names. From the plugin usage:

```
-a, --authproto=STRING      SNMPv3 auth proto (MD5|SHA)
```

Reply:
Ah! You meant "MD5|SHA"! Fine.

(And you're wondering about https://github.com/Icinga/icinga2/pull/8400#pullrequestreview-519444745 😉)